media/CircleGraph/index.js

```diff
@@ -147,6 +147,9 @@ host.BrowserHost = class {
     });
 
     this.window.addEventListener("message", (event) => {
+      if (window.origin !== event.origin) {
+        return;
+      }
```

Comment:
I just applied it empty-minded. 
This one makes circle viewer not working. 
Let me know if anyone knows the right way.